core/src/main/java/com/linecorp/armeria/internal/logging/ContentPreviewingUtil.java

```diff
@@ -125,16 +155,15 @@ protected HttpObject filter(HttpObject obj) {
 
             @Override
             protected void beforeComplete(Subscriber<? super HttpObject> subscriber) {
-                if (responseContentPreviewer != null) {
-                    ctx.logBuilder().responseContentPreview(responseContentPreviewer.produce());
-                } else {
-                    // Call requestContentPreview(null) to make sure that the log is complete.
-                    ctx.logBuilder().responseContentPreview(null);
-                }
+                produceResponseContentPreview(responseContentPreviewer);
```

Comment:
Is there any chance that `subscriber.onComplete()` and `subscription.cancel()` are called concurrently?

Reply:
Maybe we could have an atomic flag, then we won't even need the `cause instanceof CancelledSubscriptionException` checks.

Reply:
I realized that if we introduce an atomic flag, a user needs to do the extra work because `beforeCancel()` and `beforeError()` can be called twice in certain situation.
So let me just remove `beforeCancel()` and handle the cancelation event in `beforeError`.